xep-0308.xml

```diff
@@ -100,11 +106,17 @@
   <p>While it's not possible to prevent this protocol from being used in such a way, it is not intended that it provides a way to continue expanding a previous message indefinitely and clients, in as much as it is sensible, should present use of this extension only for correction rather than for providing a continuous stream, for which &xep0301; can be used instead.</p>
   <p>Correction MUST only be used to change the details of a stanza (e.g. the message body) and not to change the nature of the stanza (e.g. correction MUST NOT be used to turn a chat message into a pubsub notification)</p>
   <p>While it is possible to use this protocol to correct messages older than the most recent received from a full JID, such use is out of scope for this document and support for this SHOULD NOT be assumed without further negotiation.</p>
+  <p>A &xep0045; room SHOULD reject any correction attempt made by a different real bare JID than the one from which the message originates.  In order to notify clients of this behaviour, it MUST include the 'urn:xmpp:message-correct:0' feature in its disco#info if it applies this rule.</p>
 </section1>
 <section1 topic='Security Considerations' anchor='security'>
   <p>The replacement message could have an entirely different meaning from the original message, so clients will need to make users aware that the displayed message has been edited. It is also suggested that clients make the original message available in some way, although this UI consideration is out of the scope of this document.</p>
   <p>There exist some payload types where correction is problematic, for some deployments. In particular, care needs to be taken that security labels from &xep0258; are handled in an appropriate manner for the security domains of a given deployment.</p>
-  <p>When used in a &xep0045; context, corrections must not be allowed (by the receiver) for messages received before the sender joined the room - particularly a full JID leaving the room then rejoining and correcting a message SHOULD be disallowed, as the entity behind the full JID in the MUC may have changed.</p>
+  <p>When used in a &xep0045; context, the receiver of a correction should check if the room advertises the 'urn:xmpp:message-correct:0' feature in its disco#info.  If it does so, the receiver can trust any message correction received through this room, be it live or from history.  Otherwise, if this feature isn’t advertised by the room, the receiver should verify each correction coming its way, for instance:</p>
+  <ul>
+    <li>If the room is non-anonymous and the bare JID of the sender of both the message and its correction is the same, allow it.</li>
+    <li>If the room is semi-anonymous and the message came live from the same occupant, allow it.</li>
+    <li>If the room is semi-anonymous and the message came from the history, from another sender, or from someone with the same nick but who left and came back, reject it.</li>
+  </ul>
```

Comment:
This seems to be lacking in normative language. Are these MUSTs? Is the "should" actually a "SHOULD"?